src/codegen/compile.ml

```diff
@@ -5795,23 +5796,102 @@ module FuncDec = struct
       compile_unboxed_const (E.add_fun_ptr env (E.built_in env reject_name)) ^^
       get_cb_index
 
+  let closures_to_self_reply_reject_callbacks env ts =
+    let reply_name = "@callback<" ^ Typ_hash.typ_hash (Type.Tup ts) ^ ">" in
+    Func.define_built_in env reply_name ["env", I32Type] [] (fun env ->
+        let (set_arr, get_arr) = new_local env "arr" in
+        message_start env (Type.Shared Type.Write) ^^
+        (* Look up closure *)
+        let (set_closure, get_closure) = new_local env "closure" in
+        G.i (LocalGet (nr 0l)) ^^
+        ClosureTable.recall env ^^
+        set_arr ^^ get_arr ^^
+        Arr.load_field 0l ^^ (* get the reply closure *)
+        set_closure ^^
+        get_closure ^^
+
+        (* Deserialize reply arguments  *)
+        Serialization.deserialize env ts ^^
+        (*G.i Drop ^^
```

Comment:
This is an idea to pass back (sufficiently small) results to the future's caller. Needs a follow-up issue.

Reply:
Written: #2679.